src/main/java/com/google/opensesame/servlets/MentorsServlet.java

```diff
@@ -60,4 +74,55 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(jsonMentors);
   }
+
+  // This function prints an error message to the mentor form page if invalid input is found.
+  public void formError(HttpServletRequest request, HttpServletResponse response, String error)
+      throws ServletException, IOException {
+    String message =
+        "<ul><li class='ml-3 display-4 font-weight-bold text-danger'>" + error + "</li></ul>";
+    response.setContentType("text/html");
+    PrintWriter pw = response.getWriter();
+    pw.print(message);
+    RequestDispatcher rd = request.getRequestDispatcher("mentor_form.html");
+    rd.include(request, response);
+    pw.close();
+    return;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+    String repoUrl = request.getParameter("inputRepo");
+    String urlPattern = "https://github.com/";
+    Pattern regEx = Pattern.compile(urlPattern);
+    Matcher inputMatcher = regEx.matcher(repoUrl);
+    if (!inputMatcher.find()) {
+      // Input is not in the format of a GitHub repo link.
+      formError(request, response, "ENTER A VALID REPO LINK");
+      return;
+    }
+
+    String repoName = repoUrl.replaceFirst("https://github.com/", "");
+    GitHub gitHub = GitHubGetter.getGitHub();
+    GHRepository inputRepo = gitHub.getRepository(repoName);
+    if (inputRepo == null) {
+      // Repo was not found on GitHub.
+      formError(request, response, "COULDN'T FIND REPO");
+      return;
+    }
+    // TODO: Send Richie the GHRepository to add to projects database
+
+    String userID = AuthServlet.getAuthorizedUser().getUserId();
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    try {
+      Entity personEntity = datastore.get(KeyFactory.stringToKey(userID));
+    } catch (EntityNotFoundException e) {
+      // TODO: Handle this exception (which should never occur if datastore is working properly)
+    }
```

Comment:
We should also talk about this in a call. I think it would be best if this logic is placed in an AuthServlet helper function, and one is going to be implemented soon.